syntax = "proto2";
package perception_feature.proto;
message BytesList {
  repeated bytes value = 1;
}
message FloatList {
  repeated float value = 1 [packed = true];
}
message DoubleList {
  repeated double value = 1 [packed = true];
}
message Int64List {
  repeated int64 value = 1 [packed = true];
}
message IntList {
  repeated int32 value = 1 [packed = true];
}
message BoolList {
  repeated bool value = 1 [packed = true];
}
//index value record
message IVRecordEntry{
  optional int32 index = 1;
  optional Record value = 2;
}
//index value record
message IVRecordList{
  repeated IVRecordEntry iv_record = 1;
}
message Record{
  optional BytesList bytes_list = 1;
  optional BoolList bool_list = 2;
  optional IntList int_list = 3;
  optional Int64List int64_list = 4;
  optional FloatList float_list = 5;
  optional DoubleList double_list = 6;
  optional IVRecordList iv_list = 7;
}
message RecordEntry{
  optional string key = 1;
  optional Record value = 2;
}

enum OutputFormat{
  OUT_RESERVED = 0;
  OUT_LIBSVM = 1;
  OUT_CSV = 2;
  OUT_KV = 3;
  OUT_IV = 4;
}
enum FeatureType{
  TYPE_RESERVED = 0;
  CONTINUOUS = 1; //连续特征
  DISCRETE = 2;//离散特征
  VECTOR=3;// 向量特征
}